libcst/metadata/scope_provider.py

```diff
@@ -584,12 +587,32 @@ class ComprehensionScope(LocalScope):
     pass
 
 
+# Generates dotted names from an Attribute or Name node:
+# Attribute(value=Name(value="a"), attr=Name(value="b")) -> ("a.b", "a")
+# each string has the corresponding CSTNode attached to it
+def _gen_dotted_names(
+    node: Union[cst.Attribute, cst.Name]
+) -> Iterable[Tuple[str, Union[cst.Attribute, cst.Name]]]:
+    if isinstance(node, cst.Name):
+        yield (node.value, node)
+    else:
+        value = node.value
+        if not isinstance(value, (cst.Attribute, cst.Name)):
+            raise ValueError(f"Unexpected name value in import: {value}")
+        name_values = iter(_gen_dotted_names(value))
+        (next_name, next_node) = next(name_values)
+        yield (f"{next_name}.{node.attr.value}", node)
+        yield (next_name, next_node)
+        yield from name_values
```

Comment:
This is duplicated to `get_full_name_for_node`. Let's reuse the helper.
https://libcst.readthedocs.io/en/latest/helpers.html#libcst.helpers.get_full_name_for_node

Reply:
I didn't read it right, so it generates all possible dotted names, e.g. a, a.b, a.b.c
This is specific for Attribute, should we just make it a helper of `Attribute`, so it's easier to use?

Reply:
To make it clear, my suggestion is:

Add a helper in Attribute and maybe name it as `get_dotted_prefix_names`  to make it clearer. https://github.com/Instagram/LibCST/blob/4544539485c0d5ab8dad776465b9597fb1e16baa/libcst/_nodes/expression.py#L1358
Then when you see a `Name` node in `ScopeVisitor`, you can just do `node.value` to get the string value.
And only do `node.get_dotted_prefix_names` when the node is an `Attribute`.

Reply:
Hmm I don't think this helper function makes sense for all `Attribute`s. For example: `a[b].c` or `(a+b).c`.
I think this only makes sense in the context of `Attribute`s inside import-like nodes.